File: solutions/1748/main.cpp
#include <vector> 
#include <algorithm> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    int sumOfUnique(vector<int>& nums) {
        sort(nums.begin(), nums.end()); 
        int result = 0, i = 0, j = 1; 
        while (i < nums.size()) {
            while (j < nums.size() && nums[j] == nums[i]) {j++;} 
            if (j - i == 1) {
                result += nums[i];
                i++; j++; 
            } 
            else {
                i = j; j++; 
            }
        }
        return result; 
    }
};

TEST(Test1748, SimpleTest) {
    vector<int> nums{1,2,3,2}; 
    Solution s; 
    EXPECT_EQ(s.sumOfUnique(nums), 4); 
}

int main() {
    testing::InitGoogleTest(); 
    return RUN_ALL_TESTS(); 
}

File: solutions/1748/main.py
from typing import List 
from collections import defaultdict

class Solution:
    def sumOfUnique(self, nums: List[int]) -> int:
        nums_dict = defaultdict(int)
        for n in nums:
            nums_dict[n] += 1 
        result = 0
        for m, n in nums_dict.items():
            if n == 1: 
                result += m
        return result  

if __name__ == "__main__": 
    nums = [1,2,3,2]
    s = Solution()
    assert s.sumOfUnique(nums) == 4 